sources/TTYIN

```diff
@@ -5608,11 +5609,19 @@ Copyright (c) 1982-1988, 1990-1991, 2021 by Venue & Xerox Corporation.
     TTYINEDIT.SCRATCH])
 
 (\TTYIN.RPEOF
-  [LAMBDA (STREAM)                                       (* lmm "14-Nov-86 17:06")
-                                                             (* ; 
-             "End of stream op for ttyin scratch file -- supplies as many closing parens as needed")
-
-    (CHARCODE ")"])
+  [LAMBDA (STREAM)                                      (* ; "Edited 16-Aug-2021 23:40 by rmk:")
```

Comment:
i don't understand this

Reply:
This sets up the TTYIN stream so that it returns infinitely many right parens as encoded in the bytes that the external format of the stream expects.  You can't simply return the natural single byte as the character, because the stream may be funky in some other way (e.g. not runcoded, in which case you need a 0 byte in front of the ) byte).  \FORMATBYTESTREAM is the new generic way of calculating the bytes that would represent any expression that you can print.